commit 8d8f2185f9b1b023a390fb5d80f10de56fc996b0
Author: Robin Murphy <robin.murphy@arm.com>
Date:   Wed Jan 25 18:31:31 2017 +0000

    arm64: dma-mapping: Fix dma_mapping_error() when bypassing SWIOTLB
    
    commit adbe7e26f4257f72817495b9bce114284060b0d7 upstream.
    
    When bypassing SWIOTLB on small-memory systems, we need to avoid calling
    into swiotlb_dma_mapping_error() in exactly the same way as we avoid
    swiotlb_dma_supported(), because the former also relies on SWIOTLB state
    being initialised.
    
    Under the assumptions for which we skip SWIOTLB, dma_map_{single,page}()
    will only ever return the DMA-offset-adjusted physical address of the
    page passed in, thus we can report success unconditionally.
    
    Fixes: b67a8b29df7e ("arm64: mm: only initialize swiotlb when necessary")
    CC: Jisheng Zhang <jszhang@marvell.com>
    Reported-by: Aaro Koskinen <aaro.koskinen@iki.fi>
    Signed-off-by: Robin Murphy <robin.murphy@arm.com>
    Signed-off-by: Will Deacon <will.deacon@arm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1299_8d8f2185f9b1b023a390fb5d80f10de56fc996b0.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1299_8d8f2185f9b1b023a390fb5d80f10de56fc996b0.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
